Modix.Data/ModixContextAutoMigrationStartupAction.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Modix.Data
+{
+    public interface IModixContextAutoMigrationStartupAction
+    {
+        Task WhenDone { get; }
+    }
+
+    public class ModixContextAutoMigrationStartupAction
+        : ScopedStartupActionBase,
+            IModixContextAutoMigrationStartupAction
+    {
+        public ModixContextAutoMigrationStartupAction(
+                ILogger<ModixContextAutoMigrationStartupAction> logger,
+                IServiceScopeFactory serviceScopeFactory)
+            : base(
+                logger,
+                serviceScopeFactory) { }
+
+        protected override async Task OnStartingAsync(
+            IServiceProvider serviceProvider,
+            CancellationToken cancellationToken)
+        {
+            ModixContextLogMessages.ContextMigrating(_logger);
+            await serviceProvider.GetRequiredService<ModixContext>().Database.MigrateAsync(cancellationToken);
```

Comment:
why is ModixContext resolved here? Why not have it passed in?

Reply:
Cause `ModixContextAutoMigrationStartupAction` is `Singleton` and `ModixContext` is `Scoped`.

Reply:
Cause ModixContextAutoMigrationStartupAction is Singleton and ModixContext is Scoped.

Reply:
You could have a ModixContext factory instead of a service provider. But I don't think it's super necessary unless you plan on adding tests around this. The change could be made at that time I guess.

Reply:
Like, just on this one line? `IModixContextFactory.CreateContext`? I see no possible purpose for that to serve.